python_modules/dagster/dagster/core/storage/event_log/sql_event_log.py

```diff
@@ -153,56 +122,74 @@ def store_asset_materialization(self, event):
-                AssetKeyTable.update()
-                .values(  # pylint: disable=no-value-for-parameter
-                    last_materialization=serialize_dagster_namedtuple(event),
-                    last_run_id=event.run_id,
-                )
-                .where(
-                    AssetKeyTable.c.asset_key == event.dagster_event.asset_key.to_string(),
-                )
+        # The AssetKeyTable contains a `last_materialization_timestamp` column that is exclusively
+        # used to determine if an asset exists (last materialization timestamp > wipe timestamp).
+        # This column is used nowhere else, and as of AssetObservation/AssetMaterializationPlanned
+        # event creation, we want to extend this functionality to ensure that assets with any event
+        # (observation, materialization, or materialization planned) yielded with timestamp
+        # > wipe timestamp display in Dagit.
+
+        # As of the following PRs, we update last_materialization_timestamp to store the timestamp
+        # of the latest asset observation, materialization, or materialization_planned that has occurred.
+        # https://github.com/dagster-io/dagster/pull/6885
+        # https://github.com/dagster-io/dagster/pull/7319
+
+        # The AssetKeyTable also contains a `last_run_id` column that is updated upon asset
+        # materialization. This column was not being used until the below PR. This new change
+        # writes to the column upon `ASSET_MATERIALIZATION_PLANNED` events to fetch the last
+        # run id for a set of assets in one roundtrip call to event log storage.
+        # https://github.com/dagster-io/dagster/pull/7319
+
+        values = self._get_asset_entry_values(event, self.has_asset_key_index_cols())
+        insert_statement = AssetKeyTable.insert().values(
+            asset_key=event.dagster_event.asset_key.to_string(), **values
+        )
+        update_statement = (
+            AssetKeyTable.update()
+            .values(**values)
+            .where(
+                AssetKeyTable.c.asset_key == event.dagster_event.asset_key.to_string(),
             )
+        )
 
         with self.index_connection() as conn:
             try:
                 conn.execute(insert_statement)
             except db.exc.IntegrityError:
                 conn.execute(update_statement)
 
+    def _get_asset_entry_values(self, event, has_asset_key_index_cols):
+        # last_materialization_timestamp is updated upon observation, materialization, materialization_planned
+        # last_run_id column is updated upon materialization and materialization planned events.
+        # See SqlEventLogStorage.store_asset_event method for more details
+
+        entry_values: Dict[str, Any] = {}
+        if has_asset_key_index_cols:
+            entry_values.update(
+                {
+                    "last_materialization_timestamp": utc_datetime_from_timestamp(event.timestamp),
+                }
+            )
```

Comment:
I think we need to skip this for the `is_asset_materialization_planned` events, right?

Reply:
I think that we should still keep it. Say we wipe the asset and then start a run that fails to materialize the asset. If we write into this column, then the asset will be viewable in the catalog and there will be an asset details page for this asset that states that a run failed to generate the asset.

Reply:
Yikes.... This feels like a bug.  At the very least, we need to add a million comments here that `materialization_timestamp` isn't what you think it is.

Do you think we can reorganize this code?

I think having some duplication in this one section seems okay, for readability, now that it's all in the same fn.

Something like:
```
if is_asset_materialization:
   values = {...}
elif is_asset_observation:
   values = {...}
elif is_asset_planned:
   values = {...}
else:
    raise some exception
```

Even if we have some boilerplate conditionals in each section based on the cols, it might be easier to spot any bugs / changes that need to happen, and can more easily add comments?